fontbe/src/hvar.rs

```diff
@@ -0,0 +1,233 @@
+where
+    T: FontWrite + Validate,
+{
+    let data = dump_table(table).map_err(|e| Error::DumpTableError {
+        e,
+        context: context.to_string(),
+    })?;
+    Ok(data.len())
+}
+
+/// Helper to collect advance width deltas for all glyphs in a font
+struct AdvanceWidthDeltas {
+    /// Variation axes
+    axes: Vec<Axis>,
+    /// Sparse variation models, keyed by the set of locations they define
+    models: IndexMap<BTreeSet<NormalizedLocation>, VariationModel>,
+    /// Glyph's advance width deltas sorted by glyph order
+    deltas: Vec<Vec<(VariationRegion, i16)>>,
+}
+
+impl AdvanceWidthDeltas {
+    fn new(global_model: VariationModel) -> Self {
+        let axes = global_model.axes().cloned().collect::<Vec<_>>();
+        let global_locations = global_model.locations().cloned().collect::<BTreeSet<_>>();
+        // using IndexMap to quickly get a ref to the first entry (in insertion order)
+        // in the global_locations() method below
+        let mut models = IndexMap::new();
+        models.insert(global_locations, global_model);
+        AdvanceWidthDeltas {
+            axes,
+            models,
+            deltas: Vec::new(),
+        }
+    }
+
+    fn global_locations(&self) -> &BTreeSet<NormalizedLocation> {
+        self.models.first().unwrap().0
+    }
+
+    fn add(&mut self, glyph: &Glyph) -> Result<(), Error> {
+        let mut advance_widths: HashMap<_, _> = glyph
+            .sources()
+            .iter()
+            .map(|(loc, src)| (loc.clone(), vec![src.width]))
+            .collect();
+        let name = glyph.name.clone();
+        let i = self.deltas.len();
+        if advance_widths.len() == 1 {
+            assert!(advance_widths.keys().next().unwrap().is_default());
+            // this glyph has no variations (it's only defined at the default location),
+            // therefore the deltas returned from VariationModel will be an empty Vec.
+            // However, when this is the first .notdef glyph we would like to treat it
+            // specially in order to match the output of fontTools.varLib.
+            // In fonttools, all master TTFs have a .notdef glyph as their first glyph; in fontc,
+            // unless the input source defines a .notdef, only a default instance is generated.
+            // And that's ok for gvar, however for HVAR the order in which regions and associated
+            // deltas are added to VariationStoreBuilder, one glyph at a time, can produce
+            // different orderings of the ItemVariationStore.VariationRegionList (newly seen
+            // regions get appended, and existing regions reused).
+            // So, to match the VarRegionList produced by fontTools, we need to make the deltaset
+            // for the first .notdef glyph similarly "dense", by copying its default instance to
+            // all other glyph locations...
+            if i == 0 && name.as_str() == ".notdef" {
```

Comment:
Instead of comparing with a string you can use the `NOTDEF` associated context that is defined on `GlyphName`, e.g. `name == GlyphName::NOTDEF`.